### DISCUSSION TITLE
How do I set up a wave source that only propagates in a single direction?

### DISCUSSION THREAD

**Discussion by fr0ger** [NanoComp/meep#2871] [Q&A]
My purpose was simple, I just wanted a wave that traveled forward (it was Gaussian with some frequency width), and I modeled the setup after the example and found that the wave front was radial.
![image](https://github.com/user-attachments/assets/4fe9e555-c788-4998-a1be-cd0271c39974)
I tried to solve it with a Gaussian beam, I set the focus point at the center of the line source, and a waist of equal length to the line source. But the result is not satisfactory.
![image](https://github.com/user-attachments/assets/02a07a71-a16f-48ed-9d0f-240f4d1e9af8)
All I want is a plane wave which the wave front is parellel to the line source. How can I get it. Any answer is appreciated！

**Comment by stevengj** [comment]
> I tried to solve it with a Gaussian beam, I set the focus point at the center of the line source, and a waist of equal length to the line source. But the result is not satisfactory.

The line source needs to be wider than the waist to capture the exponential tails of the Gaussian beam.  The simplest choice is to make the line source extend for the whole width of your computational cell, regardless of the Gaussian beam waist.

> All I want is a plane wave which the wave front is parellel to the line source. How can I get it. Any answer is appreciated！

A planewave corresponds to an "infinite width" Gaussian beam — it is mathematically impossible to excite a planewave with a finite line source.  Any finite-length line source, or any finite-width Gaussian, will produce waves that spread out, a process called "diffraction".

There are two ways to excite a true planewave:
* You can excite a planewave within a *finite region only* by *enclosing* that region by sources, using something called the Equivalence Principle.   This is described e.g. in [our review article](https://arxiv.org/abs/1301.5366).

* You can excite a planewave with an infinite line source.  Computationally, this can be done using periodic boundary conditions, which can still be combined with PML to absorb outgoing scattered waves.  See [this Meep tutorial](https://meep.readthedocs.io/en/latest/Perfectly_Matched_Layer/#planewave-sources-extending-into-pml).

**Reply by fr0ger** [reply]
Thanks so much. I take the ideal case of plane waves for granted. I will study the documentation you provided carefully.
